src/Microsoft.AspNet.Mvc.ModelBinding/Metadata/IModelMetadataBindingDetailsProvider.cs

```diff
@@ -0,0 +1,12 @@
+﻿// Copyright (c) Microsoft Open Technologies, Inc. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using Microsoft.Framework.Internal;
+
+namespace Microsoft.AspNet.Mvc.ModelBinding
+{
+    public interface IModelMetadataBindingDetailsProvider : IModelMetadataDetailsProvider
+    {
+        void GetBindingDetails([NotNull] ModelMetadataBindingDetailsContext context);
```

Comment:
Don't like the idea of returning from here - this is a multi-provider system, and having the 'target' be irreplaceable simplifies the API.